milli/src/search/new/mod.rs

```diff
@@ -442,6 +448,34 @@ pub fn execute_search(
 
     let fields_ids_map = ctx.index.fields_ids_map(ctx.txn)?;
 
+    let docids = match vector {
+        Some(vector) => {
+            // return the nearest documents that are also part of the candidates.
+            let mut searcher = Searcher::new();
+            let hnsw = ctx.index.vector_hnsw(ctx.txn)?.unwrap_or_default();
+            let ef = hnsw.len().min(100);
+            let mut dest = vec![Neighbor { index: 0, distance: 0 }; ef];
+            let vector = normalize_vector(vector.clone());
+            let neighbors = hnsw.nearest(&vector, ef, &mut searcher, &mut dest[..]);
+
+            let mut docids = Vec::new();
+            for Neighbor { index, distance: _ } in neighbors.iter() {
+                let index = BEU32::new(*index as u32);
+                let docid = ctx.index.vector_id_docid.get(ctx.txn, &index)?.unwrap().get();
+                if universe.contains(docid) {
```

Comment:
I don't understand this check :sweat_smile:

Reply:
This is a necessary check because we don't want to leak documents hidden by filters (and tenant token filters). The HNSW can't filter when it dives, so we must do it afterward. However, it can return fewer documents than the requested amount, but it's the only quick solution I found. The other solution would be designing a predicate system on the HNSW crate.

Reply:
Fixed now, thanks Louis ❤️